src/Util/Str.php

```diff
@@ -14,20 +14,249 @@
 class Str
 {
     /**
-     * Create a slug out of the given string.
+     * Function to create a sanitized slug ready to use.
      *
-     * Non-alphanumeric characters are converted to hyphens.
+     * Slug is Unicode compatible but unicode chars are URL encoded by default.
+     * The urldecode() function can be used to decode slug when needed.
      *
-     * @param string $str
-     * @return string
+     * @param string $title The string used to generate the slug (usually post title).
+     * @return string The sanitized slug with unicode support.
      */
-    public static function slug($str)
+    public static function slug($title)
     {
+        // Replace accents and create sanitized slug
+        $slug = (new self)->sanitize_slug_string((new self)->convert_accents($title));
```

Comment:
To use that maybe i have to make all methods of the class static, right? I tryed your code directly and returns an error:

> Deprecated: Non-static method Flarum\Util\Str::sanitize_slug_string() should not be called statically in C:\xampp\htdocs\Str_dev2.php on line 30

I have no idea about how to fix this, i'm not a PHP developer sorry 🤔. If you can modify the code to make it works properly will be good.